backport-2.1: storage: delay manual splits that would result in more snapshots

tbg opened the pull request:
Backports #32594. This didn't apply cleanly, but only because I never
cherry-picked #32594 and #32594 refers a variable introduced within.

Fixes #32898.
Fixes #32900.
Fixes #32895.

/cc @cockroachdb/release

Release note (bug fix): resolve a cluster degradation scenario that
could occur during IMPORT/RESTORE operations, manifested through a
high number of pending Raft snapshots.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/33015)
<!-- Reviewable:end -->

tbg committed:
backport-2.1: storage: delay manual splits that would result in more snapshots

Backports #32594. This didn't apply cleanly, but only because I never
cherry-picked #32594 and #32594 refers a variable introduced within.

Fixes #32898.
Fixes #32900.
Fixes #32895.

/cc @cockroachdb/release

Release note (bug fix): resolve a cluster degradation scenario that
could occur during IMPORT/RESTORE operations, manifested through a
high number of pending Raft snapshots.

petermattis reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/33015#-:-LTTOhxQ48lYDvev7gQ4:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->
